linode/manager
# build: [M3-5912] - Vite

## Description 📝

- Uses Vite for
  - **Cloud Manager** dev server and production builds
  - **Storybook** dev server and production builds
  - **Cypress** Tests (preprocessor)
 - Updates Storybook v7 (needed for Vite support)
 - Improves Storybook branding, theming, and dark mode support
 - Makes Cypress use a Vite preprocessor
   - Doing this so things like `import.meta.env` work
 
## Todo ☑️
- [x] Reintroduce New Relic
  - Maybe use the `useScript` hook or just put it in the index.html and let it run in dev?
- [x] Fix SSH key fingerprints
  - We'll probably need some kind of crypto library. I'm unsure if the native web crypto functions can do what we need 
- [x] Fix algolia search
- [x] Get Cypress tests to run (works but may be unstable)
- [x] Get Storybook to run
  - Options include update to Storybook 7 and use vite builder or somehow define `import.meta.env` with current setup

## Observations 👁️
- Cypress e2e tests run stably
  - They seem to take a bit more time to run compared to `develop`. I am pretty sure this is due to the fact that with vite development mode, loading the page can actually take a bit more time due to all of the modules the browser must load (vite works like this by design)
  - When we start e2e testing with our internal CI, maybe we will benefit from doing a production build of CLoud Manager, serving it with `serve` and testing against that. This may be faster... cc @jdamore-linode 
- We should work to get our unit testing up to date. For now I am using swc to staticly define `import.meta.env` just to the tests can run. This is fine for the most part because unit tests really shouldn't need access to any environment variables

## How to test 🧪
- Checkout this PR
- Before you run anything, I'd recommend running `yarn` from the repo's root just to get new dependencies installed and old ones removed
- Run full suite of e2e tests with `yarn cy:run`
- Generally test cloud manager's dev server using `yarn up` or `yarn dev`
- Verify a Cloud Manager production build works
  - If an internal preview link is generated, it is safe to assume this works
- Test storybook dev server with `yarn storybook`
- Test storybook production build with `yarn build-storybook` (run from `packages/manager`)
